### PR TITLE
use different int8 mfma instructions on different GPUs.

### PR DESCRIPTION
**Pull request by scxiao**
Different GPU uses different mfma int8 instruction, so need changes enable that.
Note that this PR is based on the mfma fp8 PR, so should be merged after that one.

### CODE REVIEW

**Comment by zhanglx13** [comment]
Is the fp8 PR referring to 357? If so, you can change the base branch of this PR to scxiao/refine_fp8_tests.

**Review by alefimov-amd — APPROVED** [review]
LGTM

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp [hunk 1]
```diff
@@ -220,8 +231,16 @@ struct DotOpMFMAConversionHelper {
         return MatrixCoreType::FP32_BF16_BF16_FP32;
       }
     }
-    if (aElemTy.isInteger(8))
-      return MatrixCoreType::INT32_INT8_INT8_INT32;
+    if (aElemTy.isInteger(8)) {
+      auto nonKDim = mfmaEncoding.getNonKDim();
+      auto kWidth = dotOpEncoding.getKWidth();
+      if ((nonKDim == 32 && kWidth == 8) || (nonKDim == 16 && kWidth == 8)) {
```
Comment (zhanglx13) [inline lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:237]: You can only check kWidth here

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp [hunk 2]
```diff
@@ -220,8 +231,16 @@ struct DotOpMFMAConversionHelper {
...
+        return MatrixCoreType::INT32_INT8_INT8_INT32_3K;
```
Comment (zhanglx13) [inline lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: Why is `_3K` appended? Is this some kind of convention from the ISA?

Comment (scxiao) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: I saw there is a `FP32_BF16_BF16_FP32_1K` before, and think that is for MI100, so I used a _3K here. Maybe change to `300`?  I am fine with either name.

Comment (zhanglx13) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: I think `_1K` refers to the faster version of bf16 instruction on MI200. I do not know how it is named. 
@binarman Any thoughts on this?

Comment (scxiao) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: How about use the name `INT32_INT8_INT8_INT32_C3` which indicate it is start from `CDNA 3`. Then we can use `3/4R` for instructions on `RDNA` hardware.

Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: `1k` cames from mfma instruction assembly name.
There are two kinds of instructions that works with bf16:
- `V_MFMA_F32_{*}BF16_1K`  available on MI200 and MI300
- `V_MFMA_F32_{*}_BF16` available on MI100, MI200 and MI300

Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:238]: Looks like there are no such difference between old int8 and new int 8 instructions, thought.

I like this idea with `INT32_INT8_INT8_INT32_C3`
We can be even more explicit and name it `INT32_INT8_INT8_INT32_CDNA3`

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp (hunk 1 above)
Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp:237]: I think keeping `nonKDim` checks will not hurt.
This is like an additional sanity check.

File: lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp (hunk 2 above)
Comment (scxiao) [reply lib/Conversion/TritonGPUToLLVM/DotOpToLLVM/MFMA.cpp]: changed to `INT32_INT8_INT8_INT32_CDNA3`.

**Pull request merged**